math_utils.py

```diff
@@ -29,3 +29,70 @@ def factorial(n):
     for i in range(1, n + 1):
         result *= i
     return result
+
+def fibonacci(n):
+    """
+    Calculates the nth Fibonacci number.
+    """
+    if n < 0:
+        raise ValueError("Fibonacci is not defined for negative numbers")
+    if n == 0:
+        return 0
+    if n == 1:
+        return 1
+    a, b = 0, 1
+    for _ in range(2, n + 1):
+        a, b = b, a + b
+    return b
+
+def is_prime(n):
+    """
+    Checks if a number is prime.
+    """
+    if n < 2:
+        return False
+    for i in range(2, int(n ** 0.5) + 1):
+        if n % i == 0:
+            return False
+    return True
+
+def gcd(a, b):
+    """
+    Calculates the greatest common divisor of two numbers.
+    """
+    while b:
+        a, b = b, a % b
+    return a
+
+def lcm(a, b):
+    """
+    Calculates the least common multiple of two numbers.
+    """
+    return a * b // gcd(a, b)
+
+def is_anagram(s1, s2):
+    """
+    Checks if two strings are anagrams.
+    """
+    return sorted(s1) == sorted(s2)
+
+def is_armstrong(n):
+    """
+    Checks if a number is an Armstrong number.
+    """
+    s = str(n)
+    return n == sum(int(c) ** len(s) for c in s)
+
```

Comment:
I'm not sure this function will work. Have you tested it?

Reply:
I'm sure it will work.